src/python/grpcio/grpc/experimental/aio/_channel.py

```diff
@@ -89,6 +89,19 @@ def __init__(
         self._response_deserializer = response_deserializer
         self._interceptors = interceptors
 
+    @staticmethod
+    def _init_metadata(metadata: Optional[Metadata] = None,
+                       compression: Optional[grpc.Compression] = None
+                      ) -> Metadata:
+        """Based on the provided values for <metadata> or <compression> initialise the final
+        metadata, as it should be used for the current call.
+        """
+        metadata = metadata or Metadata()
+        if compression:
+            metadata = Metadata(
+                *_compression.augment_metadata(metadata, compression))
```

Comment:
Most likely Im missing something, but I would say that `augment_metadata` is only dealing with tuples, not `Metadata` objects. If the first parameter is None, automatically this function will receive as a first parameter an empty `Metadata`.

Should we just first make the augmentation - new tuple created - and then, later on, create the metadata object?

am I wrong?

Reply:
Yes, but ``augment_metadata``, internally does ``tuple(metadata)``, which works for the metadata object, so that remains as before.